universal-application-tool-0.0.1/app/forms/QuestionForm.java

```diff
@@ -114,10 +98,20 @@ public QuestionDefinitionBuilder getBuilder() {
         new QuestionDefinitionBuilder()
             .setQuestionType(questionType)
             .setName(questionName)
-            .setPath(getQuestionPath())
+            .setPath(path)
             .setDescription(questionDescription)
             .setQuestionText(questionTextMap)
             .setQuestionHelpText(questionHelpTextMap);
     return builder;
   }
+
+  /** This is a temporary fix until Path is properly handled. */
+  public Path getPath() {
```

Comment:
question: why is this public? Can't we just call `getPath` in `getBuilder`?

Reply:
I wanted `getBuilder` to take a path argument, since we'll need to be passing a path somehow. One alternative is `getBuilder` returns a builder that doesn't have path set, and `builder.setPath(path)` needs to be called afterwards each time. I think it makes more sense to pass it in as an argument.

Right now, the `QuestionForm#getPath` method is used, but as soon as repeated questions are added the paths will be coming from `QuestionService#makePath`.

Reply:
Does that mean we'll remove `QuestionForm#getPath` once we start using `ReadOnlyQuestionService#makePath`? I don't fully understand why we don't use that now since it already exists - think I'm just missing context from the complexity of repeater(d)s. :)

Reply:
Agree that passing as an argument makes more sense since callers can't forget to set it

Reply:
The `QuestionForm` doesn't have access to the `ReadOnlyQuestionService` so it needs to be passed in as an argument.

Reply:
Oh, cool. #781 is helpful context!